File: sdk/go/quicklink.go
package quicklink
import("bytes";"encoding/json";"fmt";"io";"net/http";"strings")
type Client struct{BaseURL,APIKey string; HTTP *http.Client}
type Error struct{Status int; Message string; RequestID string}
func(e *Error) Error()string{return fmt.Sprintf("quicklink api %d: %s",e.Status,e.Message)}
func(c *Client) do(path,method string,body any)(map[string]any,error){var b []byte;var err error;if body!=nil{b,err=json.Marshal(body);if err!=nil{return nil,err}};req,err:=http.NewRequest(method,strings.TrimRight(c.BaseURL,"/")+path,bytes.NewReader(b));if err!=nil{return nil,err};req.Header.Set("Authorization","Bearer "+c.APIKey);req.Header.Set("Accept","application/json");req.Header.Set("Content-Type","application/json");hc:=c.HTTP;if hc==nil{hc=http.DefaultClient};res,err:=hc.Do(req);if err!=nil{return nil,err};defer res.Body.Close();raw,readErr:=io.ReadAll(io.LimitReader(res.Body,2<<20));if readErr!=nil{return nil,readErr};if res.StatusCode>=300{var e struct{Error string `json:"error"`;RequestID string `json:"requestId"`};_ = json.Unmarshal(raw,&e);return nil,&Error{Status:res.StatusCode,Message:e.Error,RequestID:e.RequestID}};out:=map[string]any{};if len(raw)>0{if err=json.Unmarshal(raw,&out);err!=nil{return nil,err}};return out,nil}
func(c *Client) Campaigns()(map[string]any,error){return c.do("/api/v1/campaigns","GET",nil)}
func(c *Client) CreateLink(url string)(map[string]any,error){return c.do("/api/shorten","POST",map[string]string{"url":url})}

File: sdk/python/quicklink.py
import json, urllib.error, urllib.request
class QuickLinkError(Exception):
    def __init__(self,status,message,request_id=None): super().__init__(message); self.status=status; self.message=message; self.request_id=request_id
class Client:
    def __init__(self, base_url: str, api_key: str, timeout: float=10): self.base_url=base_url.rstrip('/'); self.api_key=api_key; self.timeout=timeout
    def request(self,path,method='GET',body=None):
        data=json.dumps(body).encode() if body is not None else None
        req=urllib.request.Request(self.base_url+path,data=data,method=method,headers={'Authorization':'Bearer '+self.api_key,'Accept':'application/json','Content-Type':'application/json'})
        try:
            with urllib.request.urlopen(req,timeout=self.timeout) as r: return json.loads(r.read() or b'{}')
        except urllib.error.HTTPError as e:
            try: b=json.loads(e.read() or b'{}')
            except Exception: b={}
            raise QuickLinkError(e.code,b.get('error',f'HTTP {e.code}'),b.get('requestId')) from e
    def campaigns(self): return self.request('/api/v1/campaigns')
    def create_link(self,url): return self.request('/api/shorten','POST',{'url':url})
